template/src/services/networking/index.ts

```diff
@@ -0,0 +1,78 @@
+import {AxiosError, AxiosRequestHeaders, AxiosResponse} from 'axios'
+import {IAxiosMethod, IAxiosError, IParams} from '../../constants/interface/services/axios'
+import {isObject} from '../../utilities/utils'
+import instance from './axios'
+
+const AxiosMethod: IAxiosMethod = {
+  get: 'GET',
+  post: 'POST',
+  put: 'PUT',
+  delete: 'DELETE',
+  patch: 'PATCH',
+}
+
+async function axiosAPI<T>(params: IParams<T>): Promise<AxiosResponse<T>> {
+  const {url, method, body, config} = params
+  let data = body
+
+  if (isObject(body)) {
+    data = JSON.stringify(body)
+  }
+  return instance({
+    url,
+    method,
+    data,
+    headers: {...config},
+  })
+    .then((response: AxiosResponse) => {
+      return {...response.data, status: response.status}
+    })
+    .catch((error: AxiosError<T>) => {
+      return {error: error?.response?.data, status: error?.response?.status}
+    })
+}
+
+export function getRequest<T>(params: IParams<T>): Promise<AxiosResponse<T>> {
+  const {url, config} = params
+  return axiosAPI<T>({url, method: AxiosMethod.get, config})
+}
+
+export function postRequest<T>(params: IParams<T>): Promise<AxiosResponse<T>> {
+  const {url, body, config} = params
+  return axiosAPI<T>({url, method: AxiosMethod.post, body, config})
+}
+
+export function postFormDataRequest<T>(params: IParams<T>): Promise<AxiosResponse<T>> | IAxiosError {
+  const {url, body, config} = params
+  try {
+    if (body?.constructor !== FormData) {
+      throw new Error('Unrecognized FormData part')
+    }
+    const headers: AxiosRequestHeaders = {
+      'content-type': 'multipart/form-data',
+    }
+    const customConfig = {
+      ...config,
+      ...headers,
```

Comment:
If this `headers` only have one value above, why don't we just add it here?

Reply:
done